src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/base/CheckboxesContainerComponent.test.tsx

```diff
@@ -303,4 +304,35 @@ describe('CheckboxContainerComponent', () => {
       container.querySelectorAll('.MuiFormGroup-root.MuiFormGroup-row').length,
     ).toBe(0);
   });
+
+  it('should present replaced label if setup with values from repeating group in redux and trigger handleDataChanged with replaced values', () => {
+    const handleDataChange = jest.fn();
+
+    const checkboxesWithOptionsFromRedux: ICheckboxContainerProps = {
+      options: [],
+      preselectedOptionIndex: undefined,
+      validationMessages: {},
+      legend: 'legend',
+      handleDataChange,
+      handleFocusUpdate: jest.fn(),
+      getTextResource: (value) => value,
+      getTextResourceAsString: (value) => value,
+      source: {
+        group: "someGroup",
+        label: "option.from.rep.group.label",
+        value: "someGroup[{0}].valueField"
+      },
+      ...({} as IComponentProps),
+    };
```

Comment:
Would it not be possible to use the `render` helper method here? Could extend it to accept a custom preloaded state as well. That would reduce some boilerplate, f.ex you would only need to pass overridden props to the component.

Reply:
Jupp, that is much better, thanks. Refactored.